wallet/src/main/java/com/android/identity_credential/wallet/ui/prompt/biometric/BiometricPrompt.kt

```diff
@@ -8,6 +8,53 @@ import androidx.biometric.BiometricPrompt.PromptInfo
 import androidx.core.content.ContextCompat
 import androidx.fragment.app.FragmentActivity
 import com.android.identity.android.securearea.UserAuthenticationType
+import com.android.identity_credential.wallet.R
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+/**
+ * [showBiometricPrompt]
+ * Async extension function of a [FragmentActivity] to show Biometric Prompt and return the result as a [Boolean] of whether
+ * authentication was successful, or raises/throws an Exception, such as [IllegalStateException],
+ * if an error prevented the Biometric Prompt from showing.
+ *
+ * @param title the title for the authentication prompt.
+ * @param subtitle the subtitle for the authentication prompt.
+ * @param cryptoObject a crypto object to be associated with this authentication.
+ * @param userAuthenticationTypes the set of allowed user authentication types, must contain at
+ *                                least one element.
+ * @param requireConfirmation option to require explicit user confirmation after a passive biometric.
+ * @return a [Boolean] indicating whether biometric authentication was successful.
+ * @throws Exception if there were errors showing the prompt
+ */
+suspend fun FragmentActivity.showBiometricPrompt(
```

Comment:
Why is this an extension of FragmentActivity but the other version with callbacks isn't?

FWIW, I think neither should be extensions of FragmentActivity.

Ditto for consent and passphrase.

Reply:
Good question, my Kotlin rule of thumb is that if you're creating a new object and you're passing `this` to it, then the creation of that object might as well occur on `this`.

The old/traditional Java ways of passing `this` are not that popular in Kotlin that is in favor of using the object directly to provide the created object and not pass around instances of "yourself".

But it's not always feasible to move the creation of the object into `this`, and so the next best thing is to provide that business logic as an extension of `this`'s functionality. This makes the code very clean and readable.

Reply:
I'm not really sure we're actually _extending_ FragmentActivity. We're just using it to anchor stuff. So it seems kinda weird to use extensions here. 

> Good question, my Kotlin rule of thumb is that if you're creating a new object

I don't think of the API as "creating a new object". That seems just to be an implementation detail.

I Googled around a bit for guidance and found [this](https://stackoverflow.com/questions/35317940/when-should-one-prefer-kotlin-extension-functions) which seems like balanced advice. Our use case seems to fall into bullet 8 (Optional cases).

Your use-case is using this from inside an Activity and I can see in that sense it makes sense for this to be an extension. But it's just as likely you'd be using this from outside, for example in `onClick` in Compose code, for example we use it to biometrically auth the user before showing Document data. In this case it's kinda awkward how it's an extension function.

So I think these shouldn't be extension functions at all. Please just make them take the activity as the first parameter.